pilot/pkg/model/push_context.go

```diff
@@ -259,18 +261,28 @@ func (first *PushRequest) Merge(other *PushRequest) *PushRequest {
 		Reason: append(first.Reason, other.Reason...),
 	}
 
-	// Only merge EdsUpdates when incremental eds push needed.
-	if !merged.Full {
```

Comment:
why do we need merge when it is full

Reply:
Because "ConfigsUpdated" will hold different kinds of changed resources. For example, with both full "PushRequest", one holds changed resource kind A(like "Namespace"), another holds B(like "ServiceEntry"), we still need to merge them into two changed resource.

Reply:
I suppose you want to do full push to specific proxies that maybe influenced?